### PR TITLE
Handle login failure when cookies are blocked or missing

### LINKED ISSUE #9283: KeyError: 'macaroon_root'
https://sentry.is.canonical.com/canonical/ubuntu-com/issues/9570/?referrer=github_plugin

```
KeyError: 'macaroon_root'
(4 additional frame(s) were not displayed)
...
  File "flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "site-packages/flask_openid.py", line 510, in decorated
    return self.after_login_func(OpenIDResponse(
  File "webapp/login.py", line 84, in after_login
    root = Macaroon.deserialize(flask.session.pop("macaroon_root"))
  File "werkzeug/datastructures.py", line 269, in pop
    rv = super(UpdateDictMixin, self).pop(key)

KeyError: 'macaroon_root'
```

### PR DESCRIPTION
**Pull request by jpmartinspt**
## Done

- Show error page when user tries to login but has blocked cookies

## QA

- Check out this feature branch
- Run the site using the command `./run serve` or `dotrun`
- View the site locally in your web browser at: http://0.0.0.0:8001/
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)
- Disable cookies on your browser for `localahost:8001`
- Try to login and see the error page

## Issue / Card

Fixes #9283

### CODE REVIEW

File: templates/templates/_error_login.html [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+          hi_def=True,
+          loading="auto"
+        ) | safe
+      }}
+    </div>
+    <div class="col-6 u-vertically-center">
+      <div>
+        <h1 class="p-heading--two">Something's gone wrong.</h1>
+        <p>Your cookie might have expired or your browser might be blocking cookies for this domain.</p>
+        <p>Please allow cookies for this domain and try again.</p>
```
Comment (nottrobin) [inline templates/templates/_error_login.html:24]: I remember seeing many errors like this on other sites in the past, and in every case I always had cookies enabled. So I don't think this is a very good message.

Since we're making a very specific check in the code when we raise this error, could we reword this to make it much more explicit so we have a hope of debugging it accurately?

How about:

```suggestion
        <p>Your browser session appears to be missing the expected authentication macaroon. This could be because cookies are disabled.</p>
```

**Review by nottrobin — APPROVED** [review]
Suggestion for a better message and error code. Once you've considered that, feel free to merge.

File: webapp/login.py [hunk 2]
```diff
@@ -78,7 +78,16 @@ def login_handler():
...
 def after_login(resp):
-    root = Macaroon.deserialize(flask.session.pop("macaroon_root"))
+    try:
+        root = Macaroon.deserialize(flask.session.pop("macaroon_root"))
+    except KeyError:
+        return (
+            flask.render_template(
+                "templates/_error_login.html",
+            ),
+            500,
```
Comment (nottrobin) [inline webapp/login.py:88]: I think a `400` is a more accurate error code.

**Comment by jpmartinspt** [comment]
@nottrobin out of curiosity did you disable all cookies or just cookies for the demo domain? I do see the page when I do it only for the domain.

**Comment by nottrobin** [comment]
> @nottrobin out of curiosity did you disable all cookies or just cookies for the demo domain? I do see the page when I do it only for the domain. Made the changes though as I think your message is more accurate anyway.

Yeah I disabled all cookies. Of course I wouldn't see it if it was just for that domains as login.ubuntu.com would still work.

I still think it's very unlikely that people are disabling cookies specifically and only for ubuntu.com. It's far more likely that something else went wrong.
